Library for beta sensor added

NoMonsters opened the pull request.

Li2s committed:
Library for beta sensor added

Li2s commented on Libraries/Beta/P3002.cpp:
```diff
@@ -0,0 +1,37 @@
+{
+	this->hadc = hadc;
+	k_adc_to_angle = 0.001221;
+	b_adc_to_angle = -2.5;
```
ну кэфы тут надо будет поправить

PugachA commented on Libraries/Beta/P3002.cpp:
```diff
@@ -0,0 +1,37 @@
+#include "P3002.h"
+#include "math.h"
```
В топку

PugachA commented on Libraries/Beta/P3002.cpp:
```diff
@@ -0,0 +1,37 @@
+
+void P3002::convertADC(void)
+{
+	HAL_ADC_Start(&this->hadc);
```
Его один раз недостаточно стартануть?

NoMonsters replied:
недостаточно! Надо каждый раз

PugachA commented on Libraries/Beta/P3002.cpp:
```diff
@@ -0,0 +1,37 @@
+
+uint32_t P3002::getRawData(void)
+{
+	convertADC();
```
Зачем записывать это в поле класса, если можно просто возвращать в методе `convertADC`? Ты ведб больше нигде это поле не используешь.

NoMonsters commented:
Ну шо кто-нибудь править будет? @Li2s

Li2s commented:
> Ну шо кто-нибудь править будет? @Li2s 
> 
> 

Поправлю. Дайте отпуск до выходных, пока не уволился с аривала

Li2s committed:
math.h excluded

Li2s committed:
Сменил void методы на числовые

convertADC записывает не в поле класса, а возвращает значение;
calcAngle записывает не в поле класса, а возвращает значение;
Переписал getRawData и getAngle под новые методы

Li2s committed:
Исправил присвоение дробного числа целочисленному типу

PugachA commented on Libraries/Beta/P3002.cpp:
```diff
@@ -0,0 +1,43 @@
+
+double P3002::calcAngle(void)
+{
+	uint32_t adc_raw = 0;
+	double angle = 0.0;
+
+	adc_raw = convertADC();
+	angle = k_adc_to_angle*double(adc_raw) + b_adc_to_angle;
+
+	return angle;
+}
```
uint32_t adc_raw = convertADC();
double angle = k_adc_to_angle*double(adc_raw) + b_adc_to_angle;

return angle;
